### PR TITLE
docs: updating life of a request

### PR DESCRIPTION
**Pull request by alyssawilk**
Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Review by phlax — APPROVED** [review]
nice! lgtm, thanks @alyssawilk

File: docs/root/intro/arch_overview/http/http_filters.rst [hunk 1]
```diff
@@ -9,6 +9,10 @@ HTTP level filter stack within the connection manager.
 Filters can be written that operate on HTTP level messages without knowledge of the underlying physical
 protocol (HTTP/1.1, HTTP/2, etc.) or multiplexing capabilities.
 
+HTTP filters can be downstream filters, associated with a given listener and doing stream processing on each
+downstream request before routing, or upstream filters, associated with a given cluster and doing stream once
+per upstream stream, processing after the router filter.
```
Comment (mattklein123) [inline docs/root/intro/arch_overview/http/http_filters.rst:14]:
```suggestion
HTTP filters can be downstream filters, associated with a given listener and doing stream processing on each
downstream request before routing, or upstream filters, associated with a given cluster and doing stream processing once per upstream request, after the router filter.
```

Up to you but I found the original wording a bit hard to parse.

File: docs/root/intro/life_of_a_request.rst [hunk 2]
```diff
@@ -454,6 +460,15 @@ The router filter is responsible for all aspects of upstream request lifecycle m
 stream allocated from the HTTP connection pool. It also is responsible for request timeouts, retries
 and affinity.
 
+The router filter is also responsible for the creation and running of the `Upstream HTTP filter <arch_overview_http_filters>`
+chain. By default, upstream filters will start running immediately after headers arrived at the router
```
Comment (mattklein123) [inline docs/root/intro/life_of_a_request.rst:464]:
```suggestion
chain. By default, upstream filters will start running immediately after headers arrive at the router
```

**Review by mattklein123 — APPROVED** [review]
Nice!

**Review by mattklein123 — APPROVED** [review]
